x-pack/plugins/uptime/public/components/functional/charts/checks_chart.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormattedMessage } from '@kbn/i18n/react';
+import { StatusData } from '../../../../common/graphql/types';
+import { getChartDateLabel } from '../../../lib/helper';
+import { UptimeSettingsContext } from '../../../contexts';
+import { getColorsMap } from './get_colors_map';
+
+interface ChecksChartProps {
+  /**
+   * The color that will be used for the area series displaying "Down" checks.
+   */
+  dangerColor: string;
+  /**
+   * The timeseries data displayed in the chart.
+   */
+  status: StatusData[];
+  /**
+   * The color that will be used for the area series displaying "Up" checks.
+   */
+  successColor: string;
+}
+
+/**
+ * Renders a chart that displays the total count of up/down status checks over time
+ * as a stacked area chart.
+ * @param props The props values required by this component.
+ */
+export const ChecksChart = ({ dangerColor, status, successColor }: ChecksChartProps) => {
+  const upSeriesSpecId = getSpecId('Up');
+  const downSeriesSpecId = getSpecId('Down');
+  const { absoluteStartDate, absoluteEndDate } = useContext(UptimeSettingsContext);
+
+  const upString = i18n.translate('xpack.uptime.monitorCharts.checkStatus.series.upCountLabel', {
+    defaultMessage: 'Up count',
+  });
+  const downString = i18n.translate(
+    'xpack.uptime.monitorCharts.checkStatus.series.downCountLabel',
+    {
+      defaultMessage: 'Down count',
+    }
+  );
+
+  return (
+    <React.Fragment>
+      <EuiTitle size="xs">
+        <h4>
+          <FormattedMessage
+            id="xpack.uptime.monitorCharts.checkStatus.title"
+            defaultMessage="Check status"
+          />
+        </h4>
+      </EuiTitle>
+      <EuiPanel>
+        <Chart>
+          <Settings xDomain={{ min: absoluteStartDate, max: absoluteEndDate }} showLegend={false} />
+          <Axis
+            id={getAxisId('checksBottom')}
+            position={Position.Bottom}
+            showOverlappingTicks={true}
+            tickFormat={timeFormatter(getChartDateLabel(absoluteStartDate, absoluteEndDate))}
+            title={i18n.translate('xpack.uptime.monitorChart.checksChart.bottomAxis.title', {
+              defaultMessage: 'Timestamp',
+              description: 'The heading of the x-axis of a chart of timeseries data.',
+            })}
+          />
+          <Axis
+            id={getAxisId('left')}
+            position={Position.Left}
+            tickFormat={d => Number(d).toFixed(0)}
+            title={i18n.translate('xpack.uptime.monitorChart.checksChart.leftAxis.title', {
+              defaultMessage: 'Number of checks',
+              description: 'The heading of the y-axis of a chart of timeseries data',
+            })}
+          />
+          <AreaSeries
+            customSeriesColors={getColorsMap(successColor, upSeriesSpecId)}
+            data={status.map(({ x, up }) => ({
+              x,
+              [upString]: up || 0,
+            }))}
+            id={upSeriesSpecId}
+            seriesType="area"
```

Comment:
just an FYI, here and below, you don't need to specify `seriesType` as those are provided as default props on the component for each specific series type.

Reply:
Resolved in https://github.com/elastic/kibana/pull/34313/commits/fd77d10729ef8530b7e38cd6835e7c7b4e404615.